### PR TITLE
Fix VPCI bugs found in integration testing for partition mode

### PR DESCRIPTION
**Pull request by zds100**

### CODE REVIEW

File: hypervisor/partition/vm_description.c [hunk 1]
```diff
@@ -35,37 +35,49 @@ static struct vpci_vdev_array vpci_vdev_array1 = {
...
 		.size = ALIGN_UP_4K(0x100UL),
-		.type = PCIM_BAR_MEM_32
+		.type = PCIBAR_MEM32
 		},
 		[5] = {
 		.base = 0UL,
-		.size = ALIGN_UP_4K(0x2000UL),
-		.type = PCIM_BAR_MEM_32
+		.size = ALIGN_UP_4K(0x800UL),
+		.type = PCIBAR_MEM32
```
[the remark this answers is not in the document]
Comment (zds100) [reply hypervisor/partition/vm_description.c]: Yes, you are right, PCIBAR_MEM32 is defined in another patch in the same PR, I can recreate the this patch to make it selfcontained or merge the 2 patches in this PR into one, which one do you prefer?

File: hypervisor/partition/vm_description.c [hunk 2]
```diff
@@ -35,37 +35,49 @@ static struct vpci_vdev_array vpci_vdev_array1 = {
...
 		},
 	  },
 	 .pdev = {
-		.bdf = PCI_BDF(0x00U, 0x01U, 0x00U),
+		.bdf = PCI_BDF(0x00U, 0x12U, 0x00U),
```
[the remark this answers is not in the document]
Comment (zds100) [reply hypervisor/partition/vm_description.c]: Can you elaborate on the PCI_BDF macro? as you have 2 conflicting comments.

[the remark this answers is not in the document]
Comment (anthonyzxu) [reply hypervisor/partition/vm_description.c]: Junjie mentioned we can use function like MACRO in this case.

File: hypervisor/partition/vm_description.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (zds100) [reply hypervisor/partition/vm_description.c]: ok, I will still create 2 patches in this PR

**Comment by wenlingz** [comment]
Ready to merge

**Comment by zds100** [comment]
Abandoned

**Pull request closed without merging**